lib/base-compiler.js

```diff
@@ -409,15 +409,15 @@ class BaseCompiler {
         execOptions.maxOutput = 1024 * 1024 * 1024;
 
         return this.runCompiler(this.compiler.exe, newOptions, this.filename(inputFilename), execOptions)
-            .then((output) => {
-                const ir = this.processIrOutput(output, filters);
+            .then(async output => {
```

Comment:
This async lambda in a then handler feels weird to me.

If it works and you are sick of making changes then it's probably fine, but I think since generateIR is now itself async it should be something like 
```
const res = await this.runCompiler(...);
const ir = await this.processIrOutput(...);
return ir.asm;
```

Reply:
Good spot! Yeah; will make that change. Thanks!